src/Platform/Microsoft.Testing.Extensions.TrxReport/Hashing/XxHash128.cs

```diff
@@ -0,0 +1,423 @@
+                goto case 0;
+            case 0:
+                Mix32Bytes(ref accLow, ref accHigh, source, source + length - 16, DefaultSecretUInt64_0, DefaultSecretUInt64_1, DefaultSecretUInt64_2, DefaultSecretUInt64_3, seed);
+                break;
+        }
+
+        return AvalancheHash(accLow, accHigh, length, seed);
+    }
+
+    private static Hash128 HashLength129To240(byte* source, uint length, ulong seed)
+    {
+        Debug.Assert(length is >= 129 and <= 240, "Length was expected to be between 129 and 240.");
+
+        ulong accLow = length * Prime64_1;
+        ulong accHigh = 0;
+
+        Mix32Bytes(ref accLow, ref accHigh, source + (32 * 0), source + (32 * 0) + 16, DefaultSecretUInt64_0, DefaultSecretUInt64_1, DefaultSecretUInt64_2, DefaultSecretUInt64_3, seed);
+        Mix32Bytes(ref accLow, ref accHigh, source + (32 * 1), source + (32 * 1) + 16, DefaultSecretUInt64_4, DefaultSecretUInt64_5, DefaultSecretUInt64_6, DefaultSecretUInt64_7, seed);
+        Mix32Bytes(ref accLow, ref accHigh, source + (32 * 2), source + (32 * 2) + 16, DefaultSecretUInt64_8, DefaultSecretUInt64_9, DefaultSecretUInt64_10, DefaultSecretUInt64_11, seed);
+        Mix32Bytes(ref accLow, ref accHigh, source + (32 * 3), source + (32 * 3) + 16, DefaultSecretUInt64_12, DefaultSecretUInt64_13, DefaultSecretUInt64_14, DefaultSecretUInt64_15, seed);
+
+        accLow = Avalanche(accLow);
+        accHigh = Avalanche(accHigh);
+
+        uint bound = (length - (32 * 4)) / 32;
+        if (bound != 0)
+        {
+            Mix32Bytes(ref accLow, ref accHigh, source + (32 * 4), source + (32 * 4) + 16, DefaultSecret3UInt64_0, DefaultSecret3UInt64_1, DefaultSecret3UInt64_2, DefaultSecret3UInt64_3, seed);
+            if (bound >= 2)
+            {
+                Mix32Bytes(ref accLow, ref accHigh, source + (32 * 5), source + (32 * 5) + 16, DefaultSecret3UInt64_4, DefaultSecret3UInt64_5, DefaultSecret3UInt64_6, DefaultSecret3UInt64_7, seed);
+                if (bound == 3)
+                {
+                    Mix32Bytes(ref accLow, ref accHigh, source + (32 * 6), source + (32 * 6) + 16, DefaultSecret3UInt64_8, DefaultSecret3UInt64_9, DefaultSecret3UInt64_10, DefaultSecret3UInt64_11, seed);
+                }
+            }
+        }
+
+        Mix32Bytes(ref accLow, ref accHigh, source + length - 16, source + length - 32, 0x4F0BC7C7BBDCF93F, 0x59B4CD4BE0518A1D, 0x7378D9C97E9FC831, 0xEBD33483ACC5EA64, 0 - seed);
+
+        return AvalancheHash(accLow, accHigh, length, seed);
+    }
+
+    private static Hash128 HashLengthOver240(byte* source, uint length, ulong seed)
+    {
+        Debug.Assert(length > 240, "Length was expected to be greater than 240.");
+
+#if NET
+        fixed (byte* defaultSecret = &MemoryMarshal.GetReference(DefaultSecret))
+#else
+        fixed (byte* defaultSecret = DefaultSecret)
+#endif
```

Comment:
In dotnet/runtime, `DefaultSecret` is always `ReadOnlySpan<byte>`. In this PR, I'm making it `byte[]` where `ReadOnlySpan<byte>` isn't available. Then, I'm using `fixed (byte* defaultSecret = DefaultSecret)` when ROS isn't available for us.